### PR TITLE
Add JSON schema validation for jobs and targets

### PR DESCRIPTION
**Pull request by pglass**

### CODE REVIEW

File: rift/api/common/resources.py [hunk 1]
```diff
@@ -45,12 +47,22 @@ def load_body(self, req, validator=None):
...
+                err_msg = self._format_validation_error_message(e)
+                self.abort(falcon.HTTP_400, err_msg)
 
         return obj
 
+    def _format_validation_error_message(self, error):
+        err_msg = "JSON validation failed"
+        if error.path:
+            err_msg += ": invalid value for key {0}".format(list(error.path))
+        if len(error.message) <= 100:
```
Comment (jmvrbanac) [inline rift/api/common/resources.py]: Why less than 100? Shouldn't this just be an else?

File: rift/api/common/resources.py [hunk 2]
```diff
@@ -45,12 +47,22 @@ def load_body(self, req, validator=None):
...
+            err_msg += " - " + error.message
```
Comment (jmvrbanac) [inline rift/api/common/resources.py]: ```' - {}'.format(error.message) ``` is preferred as it'll properly cast the contents to a string.

**Comment by jmvrbanac** [comment]
Overall, it looks pretty good. Could you possibly add a test or two to cover the new code?

**Comment by pglass** [comment]
I fixed the things you mentioned, and added some tests. Specter is pretty nice! I just tested the validators themselves though, and not the `on_post()` methods... I was wondering if you had thoughts on how to unittest the API itself, or if there was a way to do that already via falcon.

**Comment by jmvrbanac** [comment]
Awesome thanks!  So there are a number of ways to test the on_xxxx() methods. You can use a wsgi testing wrapper such as webtest or similar. Or because Falcon is quite straight forward and works off the passed in object references, you can create stub objects for req and resp and pass those into the on_xxxx() methods then assert on the assigned data.

**Comment by jmvrbanac** [comment]
I'll merge this for now, but it would be good to get the coverage up on those resources.

**Pull request merged**